### DISCUSSION TITLE
How do I install additional nodes?

### DISCUSSION THREAD

**Discussion by MariKiwi** [pufferpanel/pufferpanel#1279] [Q&A]
I was wondering how to install a new Node on a separate server that doesn't have the panel on it. I do know how to add a node to the panel on the website, but I don't know how to install it on the server without having to install everything?

**Comment by storm37000** [comment]
I tried downloading the linux amd64 zip and just ./pufferpanel and it says
```
group: unknown group pufferpanel
You do not have permission to use this command
```
Seems like they dont have a lightweight, http server removed, binary for nodes yet that can run under any user and stay contained within that users home folder. (i refuse to use docker with its overhead, id rather just run a node per user and seperate it the native linux way)

**Reply by nepcore** [reply]
A "lightweight, http server removed, binary" would simply not function, the master has to be able to talk to nodes in some way and this is implemented using a RESTful API, that needs an HTTP server.  
Just reusing the same package since it'd be the same code running on a node and master anyways is both cutting down the work needed to maintain everything and making it easier for less technically inclined users. And given that the full package is ~70 MB uncompressed with the majority being the binary I don't see any need to worry about the web assets existing with it.

Nodes are also not an isolation feature, if you need isolation you should really just use Docker, they're in the public eye and get some degree of auditing even should they not want it. We do our best to keep everything secure, but there isn't anywhere near that amount of eyes on our code double, triple and quadruple checking every line.  
And regarding the overhead: Running multiple PufferPanel instances has overhead too, as much as all but one of those instances use in resources combined. Docker isn't some ball and chain shackled to a processes feet and slowing it down by a factor of 100.

Regardless of that, a group is not a user and you can absolutely run the PufferPanel binary under any user in any directory, the user just needs to be part of a group called `pufferpanel`, so even your one node per user setup would be possible, though needing a bit of manual work due to being nowhere near the intention with which we built PufferPanel.

**Reply by storm37000** [reply]
that is fair, I'm more concerned about CPU overhead/stability with garrysmod SRCDS as this is just a VPS with plenty of ram and with memory compression enabled (multiple instances of a program should compress much of their memory away) that me and some friends are sharing.  Not sure how well docker will play with already being inside a VM.  I'm mostly concerned security wise about server plugins/addons reading other users stuff and the login for the front end panel being secure (might use my steam login OAUTH just to be safe).

**Comment by nepcore** [comment]
After creating a node on the panel you go to that node on the panel (currently not quite perfect, but that'll get improved in the future) and follow the instructions at the bottom of that page.  
"install it on the server without having to install everything" is not a thing, PufferPanel is ~70 MB on disk when installed, most of which is what you actually need for the node anyways and the config the panel tells you to place on the new node takes care of disabling the panel on that node, at that point the parts that aren't needed for the node consume a few MB disk space and nothing else.
